### PR TITLE
SOC-1837 | Opening discussion post in a new tab on a mouse wheel click

### PR DESCRIPTION
**Pull request by Bloodmoon24**
https://wikia-inc.atlassian.net/browse/SOC-1837
@kvas-damian  @bkoval  please CR

### CODE REVIEW

File: front/templates/main/discussion/forum.hbs [hunk 1]
```diff
@@ -3,7 +3,6 @@
 	create=(action 'create')
 	deletePost=(action 'deletePost')
 	goToAllDiscussions=(action 'goToAllDiscussions')
-	goToPost=(action 'goToPost')
```
Comment (kvas-damian) [inline front/templates/main/discussion/forum.hbs]: Please also remove https://github.com/Wikia/mercury/blob/SOC-1837/front/templates/main/components/discussion-forum.hbs#L40

File: front/scripts/main/mixins/discussion-parsed-content.js [hunk 2]
```diff
@@ -28,11 +34,13 @@ export default Ember.Mixin.create({
...
 		escapedContent = nl2br(escapedContent);
 
-		return window.Autolinker ? window.Autolinker.link(escapedContent, {
-			stripPrefix: false,
-			email: false,
-			phone: false,
-			twitter: false
-		}) : escapedContent;
+		if (!this.get('isDetailsView')) {
+			this.autolinkerConfig.replaceFn = (autolinker, match) => {
```
Comment (bkoval) [inline front/scripts/main/mixins/discussion-parsed-content.js]: Seems like the method itself could live as a separate method in the class, for clarity.

File: front/templates/main/components/post-detail.hbs [hunk 3]
```diff
@@ -21,11 +21,11 @@
 		{{{parsedContent}}}
 	</div>
 {{else}}
-	<div class="discussion-content clickable" onclick={{action 'goToPost' postId}}>
+    <div class="discussion-content">
```
Comment (kvas-damian) [inline front/templates/main/components/post-detail.hbs]: ```discussion-content``` class has styles related to text, while we have ```h2``` inside, please adjust styles like ```-webkit-line-clamp```

File: front/scripts/main/mixins/discussion-parsed-content.js [hunk 4]
```diff
@@ -28,11 +34,13 @@ export default Ember.Mixin.create({
...
+				if (match.getType() === 'url') {
+					return `<span class='link'>${match.getUrl()}</span>`;
```
Comment (bkoval) [inline front/scripts/main/mixins/discussion-parsed-content.js]: Technically it's not a link (it's not clickable). How about giving it a name that doesn't suggest clickability, like "url"?

File: front/styles/main/module/_discussion.scss [hunk 5]
```diff
@@ -68,16 +64,24 @@ body.discussions {
...
+			color: $discussion-content-color;
+			display: block;
+		}
 		.post-title {
 			line-height: $line-height-smallest;
 			margin: 0;
 		}
 	}
 
+	.post-title ~ .post-content {
```
Comment (kvas-damian) [inline front/styles/main/module/_discussion.scss]: Do we need this style when we have div wrapping title and content?

File: front/styles/main/module/_discussion.scss [hunk 6]
```diff
@@ -68,16 +64,24 @@ body.discussions {
...
+		margin-top: 0;
```
Comment (bkoval) [inline front/styles/main/module/_discussion.scss]: Is that needed for the mousewheel functionality?

File: front/templates/main/components/post-detail.hbs [hunk 7]
```diff
@@ -21,11 +21,11 @@
...
 		{{#if post.title}}
-			<h2 class="post-title"><{{post.title}}</h2>
+			<h2 class="post-title">{{#link-to 'discussion.post' postId class="post-details-link"}}{{post.title}}{{/link-to}}</h2>
 		{{/if}}
-		{{{parsedContent}}}
+		{{#link-to 'discussion.post' postId class="post-details-link"}}{{{parsedContent}}}{{/link-to}}
 	</div>
```
Comment (bkoval) [inline front/templates/main/components/post-detail.hbs]: Indentation inside this div is too big. Also, the closing tag has too big indentation.

File: front/styles/main/module/_discussion.scss (hunk 5 above)
Comment (Bloodmoon24) [inline front/styles/main/module/_discussion.scss]: Thanks, I forgot to remove this.

File: front/styles/main/module/_discussion.scss (hunk 6 above)
Comment (Bloodmoon24) [inline front/styles/main/module/_discussion.scss]: Thanks, I forgot to remove this.

File: front/scripts/main/mixins/discussion-parsed-content.js (hunk 4 above)
Comment (Bloodmoon24) [inline front/scripts/main/mixins/discussion-parsed-content.js]: That's make sense

File: front/templates/main/discussion/forum.hbs (hunk 1 above)
Comment (Bloodmoon24) [inline front/templates/main/discussion/forum.hbs]: Thanks!

File: front/templates/main/components/post-detail.hbs (hunk 7 above)
Comment (Bloodmoon24) [inline front/templates/main/components/post-detail.hbs]: If its still could you fix it for me? My IDE can't see that.

File: front/scripts/main/mixins/discussion-parsed-content.js [hunk 8]
```diff
@@ -28,11 +28,32 @@ export default Ember.Mixin.create({
...
 			email: false,
 			phone: false,
+			stripPrefix: false,
 			twitter: false
-		}) : escapedContent;
-	})
+		};
+
+		if (!this.get('isDetailsView')) {
+			this.autolinkerConfig.replaceFn = this.wrapInSpan;
```
Comment (bkoval) [inline front/scripts/main/mixins/discussion-parsed-content.js]: :+1:

File: front/styles/main/module/_discussion.scss [hunk 9]
```diff
@@ -68,10 +64,13 @@ body.discussions {
 	}
 
 	.discussion-content {
-		color: $discussion-content-color;
 		margin: 5px 0;
 		padding: .3rem 0;
 
+		.post-details-link{
```
Comment (bkoval) [inline front/styles/main/module/_discussion.scss]: Space before the curly bracket :)

File: front/templates/main/components/post-detail.hbs [hunk 10]
```diff
@@ -21,11 +21,11 @@
 		{{{parsedContent}}}
 	</div>
 {{else}}
-	<div class="discussion-content clickable" onclick={{action 'goToPost' postId}}>
+	<div class="discussion-content">
```
Comment (bkoval) [inline front/templates/main/components/post-detail.hbs]: Indentation looks fine now, must've just been github's weirdness ;>

File: front/styles/main/module/_discussion.scss (hunk 9 above)
Comment (Bloodmoon24) [inline front/styles/main/module/_discussion.scss]: Done

**Comment by bkoval** [comment]
LGTM

**Comment by kvas-damian** [comment]
LGTM

File: front/scripts/main/mixins/discussion-parsed-content.js [hunk 11]
```diff
@@ -28,11 +28,32 @@ export default Ember.Mixin.create({
...
+		}
+		this._super();
+	},
+
+	/**
+	 * Wraps links in span instead of anchor tag in discussion forum view
```
Comment (rogatty) [inline front/scripts/main/mixins/discussion-parsed-content.js]: Can you elaborate here on the reason for doing that?

Comment (Bloodmoon24) [inline front/scripts/main/mixins/discussion-parsed-content.js]: There is a problem with preventing default mouse wheel click action in Firefox, so I had to change the structure of post. We want to open in new tab post details view not link which is inside. Now you can use ctrl/cmd+click, wheel click and even right mouse click which was impossible on ember action. It's wrapped in span just to colorize it.

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**